src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -54,6 +58,28 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
         return commandResult;
     }
 
+    /**
+     * Determines the type of {@code Command} that is being executed and passes the appropriate arguments to it.
+     * <p>
+     * Note: When more types of commands are added to the application, add them in this method accordingly.
+     * <p/>
+     *
+     * @param command The command to be executed.
+     * @return The {@code CommandResult} from executing the command.
+     * @throws CommandException if the command does not belong to any of the command groups.
+     */
+    private CommandResult execute(Command command) throws CommandException {
+        if (command instanceof MiscCommand) {
+            return ((MiscCommand) command).execute();
+        } else if (command instanceof ModelCommand) {
+            return ((ModelCommand) command).execute(model);
+        } else if (command instanceof StorageCommand) {
+            return ((StorageCommand) command).execute(storage);
+        } else {
+            throw new CommandException(MESSAGE_UNKNOWN_COMMAND);
+        }
+    }
+
```

Comment:
Interesting idea of splitting Command into three subclasses! Some suggestions -- could we replace the else-if statements into if-statements (use guard clauses instead of nesting). I think this will improve readability slightly.

Reply:
Sounds good, I'll get to it.

Reply:
Done!
![image](https://user-images.githubusercontent.com/95712150/193839720-bb061aa9-8a1c-4cf0-8038-62d7d4033ed0.png)